src/api/utils.ts

```diff
@@ -0,0 +1,20 @@
+export const windowLoaded = new Promise((accept, reject) => {
+    if (typeof window === 'undefined') {
+        return accept()
+    }
+
+    if (typeof window.addEventListener !== 'function') {
+        return reject(new Error('expected to be able to register event listener'))
+    }
+
+    window.addEventListener(
+        'load',
+        function loadHandler(event) {
+            window.removeEventListener('load', loadHandler, false)
```

Comment:
From https://developer.mozilla.org/en-US/docs/Web/API/Document/DOMContentLoaded_event

```
The DOMContentLoaded event fires when the initial HTML document has been completely loaded and parsed, without waiting for stylesheets, images, and subframes to finish loading.
```

So if MM doesn't inject yet then this will throw

Reply:
Yep, while MM is generally fast, eg.. Safe would wait till load + 30ms (back then)

Reply:
But then you are relying on the fact that "by the time you finish loading the CSS and images resources" you have the web3 injected. This doesn't need to be necessarily true, note that it can be faster for example if the content is cached.

There could be race conditions, right?